### PR TITLE
Add cache flush bit to resource records.

### PR DESCRIPTION
**Pull request by cecille**
See https://datatracker.ietf.org/doc/html/rfc6762#section-10.2.
Currently, this flag is never set by the code. Just adding as
an option.

#### Problem
Mdns defines a cache flush bit that isn't in the current implementation.

#### Change overview
Adds ability to set cache flush bit for resource records. Does not actually set any cache flush bits currently (no behaviour change)

#### Testing
Manual test on M5 all-clusters-app. Forced cache-flush bits to on, checked parsed mdns records in wireshark for correct cache-flush bit settings.

### CODE REVIEW

**Review by andy31415 — COMMENTED** [review]
Please add a unit test for this as well - validate that the cache flush bit is correctly set depending on how the SetCacheFlush is configured.

File: src/lib/mdns/minimal/records/ResourceRecord.h [hunk 1]
```diff
@@ -48,6 +48,13 @@ class ResourceRecord
         return *this;
     }
 
+    ResourceRecord & SetCacheFlush(bool set)
+    {
+        mCacheFlush = set;
+        return *this;
+    }
+    bool GetCacheFlush() { return mCacheFlush; }
```
Comment (andy31415) [inline src/lib/mdns/minimal/records/ResourceRecord.h:56]: nit: should be `const`

**Review by jmartinez-silabs — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

File: src/lib/mdns/minimal/records/ResourceRecord.h [hunk 2]
```diff
@@ -62,6 +69,7 @@ class ResourceRecord
     QType mType;
     uint32_t mTtl = kDefaultTtl;
     FullQName mQName;
+    bool mCacheFlush = false;
```
Comment (tcarmelveilleux) [inline src/lib/mdns/minimal/records/ResourceRecord.h:72]: I am surprised to see the cache flush bit be added, but in such a way that the mDNS client does not invalidate any cached records if seen. Is this something that will come in a different change?

Comment (andy31415) [reply src/lib/mdns/minimal/records/ResourceRecord.h:72]: the minmdns client is not a full mdns server so it has no "cache". Platform mdns would handle this clear outside of chip.

is this what you are referring to?

File: src/lib/mdns/minimal/records/ResourceRecord.h (hunk 1 above)
Comment (cecille) [reply src/lib/mdns/minimal/records/ResourceRecord.h]: done.

File: src/lib/mdns/minimal/records/ResourceRecord.h (hunk 2 above)
Comment (cecille) [reply src/lib/mdns/minimal/records/ResourceRecord.h:72]: The original intent of this change was to allow us to request cache flushes. It was missing in the general record, so I added it. The implementation we have currently doesn't support much with respect to caching. We maintain some info about commissionable nodes outside of the minimal implementation, but clear that on every new request regardless.

**Comment by cecille** [comment]
Unit tests added.

**Review by woody-apple — APPROVED** [review]

**Pull request merged**